shop/src/pages/admin/settings/legal/Edit.js

```diff
@@ -0,0 +1,321 @@
+            { name: 'forms', items: ['Button', 'ImageButton'] },
+            '/',
+            {
+              name: 'basicstyles',
+              items: [
+                'Bold',
+                'Italic',
+                'Underline',
+                'Strike',
+                'Subscript',
+                'Superscript',
+                '-',
+                'CopyFormatting',
+                'RemoveFormat'
+              ]
+            },
+            {
+              name: 'paragraph',
+              items: [
+                'NumberedList',
+                'BulletedList',
+                '-',
+                'Outdent',
+                'Indent',
+                '-',
+                'Blockquote',
+                'CreateDiv',
+                '-',
+                'JustifyLeft',
+                'JustifyCenter',
+                'JustifyRight',
+                'JustifyBlock',
+                '-',
+                'Language'
+              ]
+            },
+            { name: 'links', items: ['Link', 'Unlink'] },
+            {
+              name: 'insert',
+              items: [
+                'Image',
+                'Table',
+                'HorizontalRule',
+                'Smiley',
+                'SpecialChar'
+              ]
+            },
+            '/',
+            {
+              name: 'styles',
+              items: ['Styles', 'Format', 'Font', 'FontSize']
+            },
+            { name: 'colors', items: ['TextColor', 'BGColor'] },
+            { name: 'tools', items: ['Maximize'] }
+          ]
+        }}
+        onChange={onChange}
+      />
+    )
+  }
+
+  return (
+    <form
+      autoComplete="off"
+      onSubmit={async (e) => {
+        e.preventDefault()
+
+        if (saving) return
+
+        setSaving(true)
+
+        try {
+          // Make sure that the request body is either a JS Object or Array. Failure to do
+          // so will produce a SyntaxError when the body is parsed by Express' body-parser
+          // module (i.e. backend). Discussion: https://github.com/expressjs/body-parser/issues/309
+          const shopPoliciesRes = await post('/shop/policies', {
+            body: JSON.stringify(policies),
+            suppressError: true
+          })
+          shopPoliciesRes ? null : console.log(`Error posting shop's policies`)
+
+          setState({ hasChanges: true })
+          setSaving(false)
+
+          dispatch({
+            type: 'toast',
+            message: (
+              <fbt desc="admin.settings.appearance.savedMessage">
+                Settings saved
+              </fbt>
+            )
+          })
+        } catch (err) {
+          console.error(err)
+          setSaving(false)
+        } finally {
+          // Notify the backend that there have been changes to the shop's config. This is done to ensure that the deploy (i.e. 'Publish Changes') button is displayed
+          // to the user
+          const shopConfigRes = await post('/shop/config', {
```

Comment:
The implementation of the /shop/policies route should be modified to take care of that. The client should not have to make this API calls.

Reply:
Sorry un-resolving since I did not see a resolution for this?

And actually, thinking more about it, I have a more fundamental question. Why do we actually need to re-publish the shop given that the way you implemented getting the policy is by making a dynamic API call? Since the policy content is dynamically fetched by the client and served by the back-end DB, we shouldn't need to re-publish the store?

Reply:
> Sorry un-resolving since I did not see a resolution for this?

I made a couple of changes to move this logic to the API code ([commit 5215d41](https://github.com/OriginProtocol/dshop/pull/988/commits/5215d41da3e57fcb57e1fc93928155bc835ed4ae)), but after reading what you say afterwards, I see your point.